frontend/src/component/common/ConstraintAccordion/ConstraintAccordionEdit/ConstraintAccordionEditBody/ResolveInput/ResolveInput.tsx

```diff
@@ -30,6 +30,29 @@ interface IResolveInputProps {
     error: string;
 }
 
+const resolveLegalValues = (
+    constraint: IConstraint,
```

Comment:
we don't need to expect the whole constraint because we only need its values. I'd use values type in the signature and pass only what we need w/o extra fluff

Reply:
all missing unit test